numpy/core/src/multiarray/mapping.c

```diff
@@ -1115,8 +1119,13 @@ array_boolean_subscript(PyArrayObject *self,
 
         NPY_END_THREADS;
 
-        NpyIter_Deallocate(iter);
+        res = NpyIter_SafeDealloc(iter, res);
```

Comment:
So this was the place why I had the thought the `NpyIter_SafeDealloc` signature accepting an existing error return might be nice. But, in the end, I am not sure its worth it; I did not really think about it too much yet. We should add one `PyErr_Occurred()` call in `NpyIter_Dealloc` without this, but that is probably almost all.

Reply:
+1 for no new API functions